assets/src/vo/date-time/duration.js

```diff
@@ -0,0 +1,567 @@
+	populateValuesFromDuration: Symbol( 'DurationPrivateMethodsPopulateValuesFromDuration' ),
+	setValues: Symbol( 'DurationPrivateMethodsSetValues' ),
+	filterValues: Symbol( 'DurationPrivateMethodsFilterValues' ),
+};
+
+/**
+ * An array of unit names for properties in the Duration object
+ * @type {string[]}
+ */
+const unitNames = [
+	'years',
+	'months',
+	'days',
+	'hours',
+	'minutes',
+	'seconds',
+	'milliseconds',
+];
+
+/**
+ * An array of derivative unit names.
+ * These are accessors that are derivatives of base units.  For instance,
+ * "weeks" ends up being a derivative (calculated from) the "days" unit.
+ * @type {string[]}
+ */
+const derivativeUnitNames = [
+	'weeks',
+];
+
+/**
+ * Where a DateTime object represents a single point in time, a Duration object
+ * represents a length of time.
+ *
+ * Durations do not have a defined beginning and end date.  They are contextless.
+ *
+ * As an example, durations are representative of something like "2 hours" and
+ * not representative of something like "between 1pm and 3pm".
+ *
+ * Internally, the Duration class here uses `moment.Duration`.  This is an
+ * abstraction loosely following the adapter pattern so that there is a common
+ * api that can be depended on if in the future the internal library is switched
+ * to something different (such as Luxon).
+ */
+export default class Duration {
+	static UNIT_YEARS = 'years';
+	static UNIT_MONTHS = 'months';
+	static UNIT_DAYS = 'days';
+	static UNIT_HOURS = 'hours';
+	static UNIT_MINUTES = 'minutes';
+	static UNIT_SECONDS = 'seconds';
+	static UNIT_MILLISECONDS = 'milliseconds';
+	static UNIT_WEEKS = 'weeks';
+
+	/**
+	 * The constructor for the Duration class.
+	 *
+	 * @param {Object|moment.Duration|string|number} values
+	 * Receiving a moment.Duration object is something for internal use and should not be used directly via
+	 * client code.
+	 * @param {string} locale  A valid locale string.
+	 * 							@link http://tools.ietf.org/html/rfc5646
+	 */
+	constructor( values, locale = DEFAULT_VALID_LOCALE ) {
+		this[ privateProperties.isValid ] = true;
+		assertions.assertLocaleIsValid( locale );
+		if ( typeof values !== 'object' ) {
+			values = moment.duration( values ).locale( locale );
+		}
+		if ( moment.isDuration( values ) ) {
+			this[ privateProperties.duration ] = values;
+			this[ privateMethods.populateValuesFromDuration ]( values );
+		} else {
+			values = this[ privateMethods.filterValues ]( values );
+			this[ privateMethods.setValues ]( values );
+			this[ privateProperties.duration ] = moment.duration(
+				values
+			).locale( locale );
+		}
+		this[ privateMethods.createGetters ]();
+		Object.freeze( this );
+	}
+
+	/**
+	 * Create an instance of Duration from a number of milliseconds.
+	 * @param {number} milliseconds
+	 * @param {string} locale
+	 * @return {Duration}  An instance of Duration.
+	 */
+	static fromMillis( milliseconds, locale = DEFAULT_VALID_LOCALE ) {
```

Comment:
how about 

```js
fromMilliseconds()
```

Reply:
I have the same reply as I did for your similar comment on the `DateTime` vo.  Luxon has `fromMillis` as well and I'd prefer to stick with following their api.